modules.d/90overlay-root/overlay-mount.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/sh
+
+# make a read-only nfsroot writeable by using overlayfs
+# the nfsroot is already mounted to $NEWROOT
+# add the parameter rootovl to the kernel, to activate this feature
+
+. /lib/dracut-lib.sh
+
+if ! getargbool 0 rootovl ; then
+    return
+fi
+
+modprobe overlay
+
+# a little bit tuning
+mount -o remount,nolock,noatime $NEWROOT
+
+# Move root
+# --move does not always work. Google >mount move "wrong fs"< for
+#     details
+mkdir -p /live/image
+mount --bind $NEWROOT /live/image
+umount $NEWROOT
+
+# Create tmpfs
+mkdir /cow
+mount -n -t tmpfs -o mode=0755 tmpfs /cow
```

Comment:
Why `-n` here?

Reply:
IIRC, at  this boot time /etc/mtab is not writeable.

Reply:
I my usecase, I always mount a read-only nfsroot.

Reply:
AFAIK / is the initrd here, so tmpfs. It's post likely linked to `/proc/self/mounts` anyway and I'd expect that the `mount --bind` calls also write to mtab.

Reply:
I'll try to check if it works without -n.

Reply:
-n can be removed, also noatime is not needed